x-pack/legacy/plugins/apm/public/components/shared/LocalUIFilters/Filter/index.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useMemo, useEffect } from 'react';
+import {
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiPopover,
+  EuiSelectable,
+  EuiSpacer,
+  EuiHorizontalRule,
+  EuiText,
+  EuiButton,
+  EuiFlexItem,
+  EuiFlexGroup
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import styled from 'styled-components';
+import { FilterBadgeList } from './FilterBadgeList';
+import { unit, px } from '../../../../style/variables';
+
+const Popover = styled(EuiPopover).attrs({
+  anchorClassName: 'anchor'
+})`
+  .anchor {
+    display: block;
+  }
+`;
+
+const SelectContainer = styled.div`
+  width: ${px(unit * 16)};
+`;
+
+const Counter = styled.div`
+  border-radius: ${theme.euiBorderRadius};
+  background: ${theme.euiColorLightShade};
+  padding: 0 ${theme.paddingSizes.xs};
+`;
+
+const ApplyButton = styled(EuiButton)`
+  align-self: flex-end;
+`;
+
+interface Props {
+  name: string;
+  title: string;
+  options: Array<{
+    name: string;
+    count: number;
+  }>;
+  onChange: (value: string[]) => void;
+  value: string[];
+  showCount: boolean;
+}
+
+const Filter: React.FC<Props> = props => {
+  const { name, title, options, onChange, value, showCount } = props;
+
+  const [showPopover, setShowPopover] = useState(false);
+
+  const toggleShowPopover = () => setShowPopover(!showPopover);
+
+  const button = (
+    <EuiButtonEmpty
+      flush="left"
+      color="text"
+      onClick={toggleShowPopover}
+      style={{ display: 'block' }}
+    >
+      <EuiTitle size="xxxs" textTransform="uppercase">
+        <h4>{title}</h4>
+      </EuiTitle>
+    </EuiButtonEmpty>
+  );
+
+  const items = useMemo(
+    () =>
+      options.map(
+        option =>
+          ({
+            label: option.name,
+            append: showCount ? (
+              <Counter>
+                <EuiText size="xs">{option.count}</EuiText>
+              </Counter>
+            ) : null,
+            ...(value.includes(option.name)
+              ? { checked: 'on' as 'on' | 'off' | 'undefined' }
```

Comment:
This looks a bit odd. Would it solve it if `onChange` on `EuiSelectable` returned items of the same type it received in `options`?

Reply:
I'm not sure what you're suggesting, is it to type the EuiSelectable.onChange callback?

In any case, this helps:
```ts
type Option = EuiSelectable['props']['options'][0];
...
const items:Option[] = ...
```

Then I don't need any additional casts. What do you think?

Reply:
Yes, I think that's better. 
What I meant is that it would be even better if EUI changed their types so that `selectedOptions` (from `onChange`) was of `typeof visibleOptions[0]`.